File: 347. Top K Frequent Elements.cpp
#include <unordered_map>
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int> dic;
        for (int i = 0; i<nums.size(); i++)
        {
            if (dic.count(nums[i]) == 0)
                dic[nums[i]] = 0;
            dic[nums[i]] += 1;
        }
        vector<pair<int, int>> vtMap;
        for (auto it = dic.begin(); it != dic.end(); it++)
            vtMap.push_back(make_pair(it->first, it->second));

        sort(vtMap.begin(), vtMap.end(), 
        [](const pair<int, int> &x, const pair<int, int> &y) -> int {
        return x.second > y.second;
        });
        
        vector<int> res;
        for (auto it = vtMap.begin(); it != vtMap.begin() + k; it++)
            res.push_back(it->first);
        return res;
    }
};

File: 347. Top K Frequent Elements.py
import operator
class Solution(object):
    def topKFrequent(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: List[int]
        """
        dic = {}
        for i in nums:
            if i not in dic:
                dic[i] = 0
            dic[i] += 1
        res = []
        for pair in sorted(dic.items(), key = operator.itemgetter(1), reverse = True):
            if len(res) == k:
                break
            res += [pair[0]]
        return res
